crates/handlers/src/compat/login_sso_complete.rs

```diff
@@ -107,6 +117,35 @@ pub async fn get(
         return Ok((cookie_jar, Html(content)).into_response());
     }
 
+    let session_counts = count_user_sessions_for_limiting(&mut repo, &session.user).await?;
+
+    let res = policy
+        .evaluate_compat_login(mas_policy::CompatLoginInput {
+            user: &session.user,
+            login_type: CompatLoginType::WebSso,
+            session_counts,
+            requester: mas_policy::Requester {
+                ip_address: activity_tracker.ip(),
+                user_agent,
+            },
+        })
+        .await?;
+    if !res.valid() {
+        let ctx = CompatLoginPolicyViolationContext::for_violations(
```

Comment:
I wonder if we should just pass the violation result? I assume this is a dependency problem, mas-templates would need to depend on mas-policy? Not a strong opinion if it's too annoying